### PR TITLE
Priority queue Audit fixes

### PR DESCRIPTION
**Pull request by pankajjagtapp**
> [!NOTE]
> **High Risk**
> Changes core withdrawal and liquidity-locking logic across `PriorityWithdrawalQueue` and `LiquidityPool`, which directly affects user funds and queue solvency. Incorrect edge-case handling (rebases, concurrent locks, or claim/cancel paths) could cause stuck funds or unintended withdrawals.
> 
> **Overview**
> **Refines priority-withdrawal semantics and solvency checks.** `PriorityWithdrawalQueue` replaces `minAmountOut` with an explicit `amountWithFee`, locks raw requested ETH on fulfillment, and enforces that the current share value can cover `amountWithFee` at claim time (with `getClaimableAmount` now returning `0` when unclaimable instead of reverting).
> 
> **Adjusts cancel/claim mechanics and remainder handling.** Cancels now return the *current* value of the originally submitted shares (rather than the originally deposited amount), finalized cancels/claims decrement the global priority lock by the raw request amount, batch-claim postcondition checks are performed per-request, and `handleRemainder` rounds the treasury split up and emits shares burned.
> 
> **Isolates LP reserves between the NFT withdrawal lock and the priority queue lock.** `LiquidityPool.withdraw` adds cross-checks so `WithdrawRequestNFT` withdrawals can’t consume ETH locked for priority withdrawals and vice versa.
> 
> **Improves ops scripts and test coverage.** Create2 deployments are made idempotent (skip-if-already-deployed with bytecode verification), upgrade transaction scripts include the `EtherFiRedemptionManager` upgrade, and tests are updated/expanded for permit frontrun behavior, rebase/slash scenarios, claimability returning zero, remainder rounding, and concurrent lock isolation.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 3b6b81bae101b8bfe24eca53b038a986f5959e74. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/PriorityWithdrawalQueue.sol [hunk 1]
```diff
@@ -19,7 +18,6 @@ import "./interfaces/IRoleRegistry.sol";
 /// @dev Implements priority withdrawal queue pattern
 contract PriorityWithdrawalQueue is 
     Initializable, 
-    OwnableUpgradeable, 
     UUPSUpgradeable, 
     ReentrancyGuardUpgradeable,
     IPriorityWithdrawalQueue
```
[the remark this answers is not in the document]
Comment (pankajjagtapp) [reply src/PriorityWithdrawalQueue.sol:23]: not yet deployed, hence safe

File: src/PriorityWithdrawalQueue.sol [hunk 2]
```diff
@@ -507,16 +500,13 @@ contract PriorityWithdrawalQueue is
         _dequeueWithdrawRequest(request);
         
         if (wasFinalized) {
-            uint256 amountForShares = liquidityPool.amountForShare(request.shareOfEEth);
```
[the remark this answers is not in the document]
Comment (pankajjagtapp) [reply src/PriorityWithdrawalQueue.sol:510]: The user is getting the appreciation of shares with positive rebase.

File: src/LiquidityPool.sol [hunk 3]
```diff
@@ -221,6 +221,8 @@ contract LiquidityPool is Initializable, OwnableUpgradeable, UUPSUpgradeable, IL
         if (totalValueInLp < _amount || (msg.sender == address(withdrawRequestNFT) && ethAmountLockedForWithdrawal < _amount) || eETH.balanceOf(msg.sender) < _amount) revert InsufficientLiquidity();
         if (_amount > type(uint128).max || _amount == 0 || share == 0) revert InvalidAmount();
 
+        if (msg.sender == priorityWithdrawalQueue && (totalValueInLp - ethAmountLockedForWithdrawal < _amount)) revert InsufficientLiquidity();
```
[the remark this answers is not in the document]
Comment (pankajjagtapp) [reply src/LiquidityPool.sol:225]: ignored

File: src/LiquidityPool.sol [hunk 4]
```diff
@@ -218,14 +219,22 @@ contract LiquidityPool is Initializable, OwnableUpgradeable, UUPSUpgradeable, IL
...
         if (_amount > type(uint128).max || _amount == 0 || share == 0) revert InvalidAmount();
 
-        totalValueInLp -= uint128(_amount);
+        if (msg.sender == priorityWithdrawalQueue && (totalValueInLp - ethAmountLockedForWithdrawal < _amount)) revert InsufficientLiquidity();
+        
         if (msg.sender == address(withdrawRequestNFT)) {
+            if (ethAmountLockedForWithdrawal < _amount) revert InsufficientLiquidity();
+            if (priorityWithdrawalQueue != address(0)) {
+                uint128 priorityLocked = uint128(IPriorityWithdrawalQueue(priorityWithdrawalQueue).ethAmountLockedForPriorityWithdrawal());
+                if (totalValueInLp - priorityLocked < _amount) revert InsufficientLiquidity();
```
[the remark this answers is not in the document]
Comment (pankajjagtapp) [reply src/LiquidityPool.sol:231]: that's alright

File: test/PriorityWithdrawalQueue.t.sol [hunk 5]
```diff
@@ -1064,24 +1292,84 @@ contract PriorityWithdrawalQueueTest is TestSetup {
...
+        priorityQueue.requestWithdraw(withdrawAmount, minAmountOut);
+
+        IPriorityWithdrawalQueue.WithdrawRequest memory request = IPriorityWithdrawalQueue.WithdrawRequest({
+            user: vipUser,
+            amountOfEEth: withdrawAmount,
+            shareOfEEth: uint96(liquidityPoolInstance.sharesForAmount(withdrawAmount)),
+            minAmountOut: minAmountOut,
+            nonce: 1,
+            creationTime: uint32(block.timestamp)
+        });
```
[the remark this answers is not in the document]
Comment (pankajjagtapp) [reply test/PriorityWithdrawalQueue.t.sol:1371]: will fix later
